frontend/views/containers/proposals/ProposalItem.vue

```diff
@@ -0,0 +1,183 @@
+<template lang='pug'>
+  li(data-test='proposalItem')
+    .c-item
+      i(:class='iconClass')
+      .c-main
+        p.has-text-bold(data-test='typeDescription') {{typeDescription}}
+        p.has-text-1(
+          :class='{ "has-text-danger": proposal.status === statuses.STATUS_FAILED, "has-text-success": proposal.status === statuses.STATUS_PASSED }'
+          data-test='statusDescription'
+          ) {{statusDescription}}
+      .ctas
+        proposal-vote-options(
+          v-if='proposal.status === statuses.STATUS_OPEN'
+          :proposalHash='proposalHash'
+        )
+    p.c-sendLink(v-if='invitationLink' data-test='sendLink')
+      i18n(
+        :args='{ user: proposal.data.proposalData.member}'
+      ) Please send the following link to {user} so they can join the group:
+      | &nbsp;
+      a.link(:href='invitationLink') {{invitationLink}}
+</template>
+
+<script>
+import { mapGetters, mapState } from 'vuex'
+import {
+  PROPOSAL_INVITE_MEMBER,
+  PROPOSAL_REMOVE_MEMBER,
+  PROPOSAL_GROUP_SETTING_CHANGE,
+  PROPOSAL_PROPOSAL_SETTING_CHANGE,
+  PROPOSAL_GENERIC,
+  STATUS_OPEN,
+  STATUS_PASSED,
+  STATUS_FAILED,
+  STATUS_EXPIRED,
+  STATUS_WITHDRAWN,
+  STATUS_CANCELLED,
+  buildInvitationUrl
+} from '@model/contracts/voting/proposals.js'
+import ProposalVoteOptions from '@containers/proposals/ProposalVoteOptions.vue'
+
+export default {
+  name: 'ProposalItem',
+  props: {
+    proposalHash: String
+  },
+  components: {
+    ProposalVoteOptions
+  },
+  computed: {
+    ...mapGetters([
+      'currentGroupState',
+      'groupMembersCount'
+    ]),
+    ...mapState(['currentGroupId']),
+    statuses () {
+      return { STATUS_OPEN, STATUS_PASSED, STATUS_FAILED, STATUS_EXPIRED, STATUS_WITHDRAWN, STATUS_CANCELLED }
+    },
+    proposal () {
+      return this.currentGroupState.proposals[this.proposalHash]
+    },
+    proposalType () {
+      return this.proposal.data.proposalType
+    },
+    typeDescription () {
+      return {
+        [PROPOSAL_INVITE_MEMBER]: () => this.L('Add {user} to group.', {
+          user: this.proposal.data.proposalData.member
+        }),
+        [PROPOSAL_REMOVE_MEMBER]: () => this.L('Remove {user} from group.', {
+          user: this.proposal.data.proposalData.member
+        }),
+        [PROPOSAL_GROUP_SETTING_CHANGE]: () => this.L('TODO: Change [setting] from [current] to [new-value]', {}),
+        [PROPOSAL_PROPOSAL_SETTING_CHANGE]: () => this.L('TODO: Change [rule setting] from [current] to [new-value]', {}),
+        [PROPOSAL_GENERIC]: () => this.L('TODO: Change [generic] from [current] to [new-value]', {})
```

Comment:
You can avoid writing `this.L` by importing the `L` function at the top (this is how other files do it). Then writing `L` instead of `this.L`